packages/amplication-data-service-generator/src/server/resource/controller/create-controller.ts

```diff
@@ -54,92 +57,13 @@ const controllerBaseTemplatePath = require.resolve(
 const toManyTemplatePath = require.resolve("./to-many.template.ts");
 
 export async function createControllerModules(
-  resource: string,
-  entityName: string,
-  entityType: string,
-  entityServiceModule: string,
-  entity: Entity,
-  srcDirectory: string
+  eventParams: CreateControllerModulesParams["before"]
 ): Promise<Module[]> {
-  const context = DsgContext.getInstance;
-  const { settings } = context.appInfo;
-  const { authProvider } = settings;
-  const entityDTOs = context.DTOs[entity.name];
-  const entityDTO = entityDTOs.entity;
-
-  const controllerId = createControllerId(entityType);
-  const controllerBaseId = createControllerBaseId(entityType);
-  const serviceId = createServiceId(entityType);
-  const createEntityId = builders.identifier("create");
-  const findManyEntityId = builders.identifier("findMany");
-  const findOneEntityId = builders.identifier("findOne");
-  const updateEntityId = builders.identifier("update");
-  const deleteEntityId = builders.identifier("delete");
-
-  const mapping = {
-    RESOURCE: builders.stringLiteral(resource),
-    CONTROLLER: controllerId,
-    CONTROLLER_BASE: controllerBaseId,
-    SERVICE: serviceId,
-    ENTITY: entityDTO.id,
-    ENTITY_NAME: builders.stringLiteral(entityType),
-    SELECT: createSelect(entityDTO, entity),
-
-    CREATE_INPUT: entityDTOs.createInput.id,
-    CREATE_DATA_MAPPING: createDataMapping(
-      entity,
-      entityDTOs.createInput,
-      DATA_ID
-    ),
-    UPDATE_INPUT: entityDTOs.updateInput.id,
-    UPDATE_DATA_MAPPING: createDataMapping(
-      entity,
-      entityDTOs.updateInput,
-      DATA_ID
-    ),
-    FIND_MANY_ARGS: entityDTOs.findManyArgs.id,
-    WHERE_INPUT: entityDTOs.whereInput.id,
-    CREATE_ENTITY_FUNCTION: createEntityId,
-    FIND_MANY_ENTITY_FUNCTION: findManyEntityId,
-    FIND_ONE_ENTITY_FUNCTION: findOneEntityId,
-    UPDATE_ENTITY_FUNCTION: updateEntityId,
-    DELETE_ENTITY_FUNCTION: deleteEntityId,
-    /** @todo make dynamic */
-    FINE_ONE_PATH: builders.stringLiteral("/:id"),
-    UPDATE_PATH: builders.stringLiteral("/:id"),
-    DELETE_PATH: builders.stringLiteral("/:id"),
-    WHERE_UNIQUE_INPUT: entityDTOs.whereUniqueInput.id,
-
-    SWAGGER_API_AUTH_FUNCTION: getSwaggerAuthDecorationIdForClass(authProvider),
-  };
-  return [
-    await createControllerModule(
-      controllerTemplatePath,
-      entityName,
-      entityType,
-      entityServiceModule,
-      entity,
-      context.DTOs,
-      mapping,
-      controllerBaseId,
-      serviceId,
-      false,
-      srcDirectory
-    ),
-    await createControllerModule(
-      controllerBaseTemplatePath,
-      entityName,
-      entityType,
-      entityServiceModule,
-      entity,
-      context.DTOs,
-      mapping,
-      controllerBaseId,
-      serviceId,
-      true,
-      srcDirectory
-    ),
-  ];
+  return pluginWrapper(
+    createControllerModulesInternal,
```

Comment:
We should use two different events 
1. createEntityController
2. createEntityConrollerBase

(like we did in the CreateService)